src/base/utils/net.cpp

```diff
@@ -47,22 +47,15 @@ namespace Utils
             return !QHostAddress(ip).isNull();
         }
 
-        Subnet parseSubnet(const QString &subnetStr, bool *ok)
+        std::optional<Subnet> parseSubnet(const QString &subnetStr)
```

Comment:
It's better to use `expected` to distinguish between regular and error cases (unless the result is really optional in regular case).

Reply:
>It's better to use expected to distinguish between regular and error cases (unless the result is really optional in regular case).

Still this discussion? The `optional` do not mean 'result/value itself is optional', it means the result/value may not be present due to whatever reasons.
See https://en.cppreference.com/w/cpp/utility/optional:
>The class template std::optional manages an optional contained value, i.e. a value that may or may not be present.
A common use case for optional is the return value of a function that may fail. 

Also see https://en.cppreference.com/w/cpp/utility/expected:
>The class template std::expected provides a way to store either of two values. An instance of std::expected at any given time either holds an expected value of type T, or an unexpected value of type E. 

In this code `E = void` but does it really fulfill the expectations? i.e. do the caller of this function expect `void` when it fails, is it meaningful? IMO the caller really expects nothing for failure cases, not even `void`, so it aligns better with `std::optional`.

Anyway, do you strongly insist to use `expected`?

Reply:
> In this code `E = void` but does it really fulfill the expectations?

No, of course. Using `expected<T, void>` is meaningless.

>i.e. do the caller of this function expect `void` when it fails? is `void` meaningful for the error case? can returning `void` be useful for the error case?

No. No. No.
>IMO the caller really expects nothing for failure cases, not even `void`, so it aligns better with `std::optional`.

IMO, it is generally better if failed function provide some error reporting instead of silently return no result.
Of course, there may be exceptions to any rule. For example, if there is obviously a single erroneous case, then you can do with `std::optional<T>`.

> Anyway, do you strongly insist to use `expected`?

No, if kind of possible error doesn't really matter.